index.js

```diff
@@ -751,6 +826,30 @@
            keys.every(function(k) { return equals(self[k], other[k]); });
   }
 
+  function unionKeys(a, b) {
+    var keys = Object.keys(a);
+    var bKeys = Object.keys(b);
+    for (var i = 0; i < bKeys.length; i += 1) {
+      if (a[bKeys[i]] === void 0) keys.push(bKeys[i]);
```

Comment:
You're right about the check being bogus. I guess the question then is should we consider _enumerable_ properties, _own_ properties, both or neither. When we're talking about a `StrMap` are we only interested in _own_ properties?